#! /bin/sh /usr/share/dpatch/dpatch-run
## 09_long_int.dpatch by  <rousseau@imac.maison.bogus>
##
## All lines beginning with `## DP:' are a description of the patch.
## DP: exif.c:405: warning: format ‘%05d’ expects type ‘int’, but argument 2 has type ‘long int’
## DP: exif.c:405: warning: format ‘%05d’ expects type ‘int’, but argument 3 has type ‘long int’
## DP: makernote.c:29: warning: format ‘%05d’ expects type ‘int’, but argument 2 has type ‘long int’
## DP: makernote.c:29: warning: format ‘%05d’ expects type ‘int’, but argument 3 has type ‘long int’

@DPATCH@
--- a/exif.c
+++ b/exif.c
@@ -477,7 +477,7 @@
             }
         }
         if (DumpExifMap){
-            printf("Map: %05d-%05d: Directory\n",DirStart-OffsetBase, DirEnd+4-OffsetBase);
+            printf("Map: %05ld-%05ld: Directory\n",DirStart-OffsetBase, DirEnd+4-OffsetBase);
         }
 
 
--- a/makernote.c
+++ b/makernote.c
@@ -26,7 +26,7 @@
         }
 
         if (DumpExifMap){
-            printf("Map: %05d-%05d: Directory (makernote)\n",DirStart-OffsetBase, DirEnd-OffsetBase);
+            printf("Map: %05ld-%05ld: Directory (makernote)\n",DirStart-OffsetBase, DirEnd-OffsetBase);
         }
     }
 
